client/src/components/DashBoard.jsx

```diff
@@ -138,7 +138,7 @@ class DashBoard extends React.Component {
   handleProfileSubmit(event) {
     event.preventDefault();
     const user = this.state.user;
-    user.UserId = this.props.user.UserId;
+    user.userId = this.props.user.userId;
      if (this.state.user.password !== null && this.state.user.passwordAgain !== null) {
           this.state.confirmOldPassword(this.props.user.email, this.state.user.oldPassword)
```

Comment:
state should hold data that will change during user interaction, and will not be changed by the reducer directly
do not pass functions to state

call this.prop.functionName directly